src/bramble/client/ProjectLimiter.js

```diff
@@ -0,0 +1,94 @@
+define(function() {
+"use strict";
+    var ProjectLimiter = {};
+
+    var maxProjectState = {
+            fileSize: 0, // max size of any file in bytes. NOT USED now.
+            fileCount: 0, // total number of files you can have in a project
+            projectSize: 0// total number of bytes for all files in this project
+    };
+    var currentProjectState = {
+        numOfFiles: 0 ,
+        projectSize: 0
+    };
+
+    ProjectLimiter.initMaxLimits =  function (projectState){
+        maxProjectState = projectState;
+        console.log(projectState);
+    };
+   function _checkLimits(){
+        if(currentProjectState.numOfFiles >= maxProjectState.fileCount || currentProjectState.projectSize >= maxProjectState.projectSize){
+            console.log("Maximum project size or file count is exceeded!\nCurrent number of files: "+currentProjectState.numOfFiles 
+            + "\nCurrent project size: " + currentProjectState.projectSize + "\nMaximum project file count: "
+            + maxProjectState.fileCount + "\nMaximum project size: " + maxProjectState.projectSize);
+        }
+    };
+
+    ProjectLimiter.newFile = function(){
+            currentProjectState.numOfFiles +=1;
+    };
+
+    //accepts path be checked, filer and the oldSize of a file.
+    //path has to be pointing to an existing file or to a file that has been just deleted. If not triggers incorrect behaviour.
+    ProjectLimiter.checkLimits = function (root,fs,oldSize){
+        var shell = new fs.Shell();
```

Comment:
As I state in the comment before, root is not pathname to a root of the project, rather a constantly changing path to a file being modified. However, you are right to suggest storing the reference to a fs in the object state.